src/client/Client.js

```diff
@@ -365,6 +366,12 @@ class Client extends BaseClient {
    * @private
    */
   _validateOptions(options = this.options) { // eslint-disable-line complexity
+    if (typeof options.ws.intents !== 'undefined') {
+      const intents = options.ws.intents;
+      if (typeof intents !== 'number' && !(intents instanceof Intents)) {
+        throw new TypeError('CLIENT_INVALID_OPTION', 'ws.intents', 'a number or Intents');
+      }
+    }
```

Comment:
I am not sure if this is the correct place or even the correct error for this, but everything else seems even more hacky. (Maybe I am overlooking something)
```suggestion
      options.ws.intents = Intents.resolve(options.ws.intents);
    }
```